src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -664,7 +664,9 @@ private IOperation CreateBoundObjectInitializerMemberOperation(BoundObjectInitia
                     }
                     else
                     {
-                        var accessor = property.GetOwnOrInheritedSetMethod();
+                        // In collection initializers, the property is not actually set. Instead, it is retrieved for a series of Add methods, so we need
+                        // to use the getter
+                        MethodSymbol accessor = inCollectionInitializer ? property.GetOwnOrInheritedGetMethod() : property.GetOwnOrInheritedSetMethod();
```

Comment:
Nested collection initializers use the Get method, as they're calling Add on the collection, rather than setting the field. If the setter was missing, it would generate an InvalidOperation.